contract_recurring_invoicing_marker/models/account_analytic_account.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# (c) 2016 Serv. Tecnol. Avanzados - Pedro M. Baeza
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import api, fields, models
+from dateutil.relativedelta import relativedelta
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = "account.analytic.account"
+
+    @api.model
+    def _prepare_invoice(self, contract):
+        next_date = fields.Date.from_string(
+            contract.recurring_next_date or fields.Date.today())
+        interval = contract.recurring_interval
+        old_date = next_date
+        if contract.recurring_rule_type == 'daily':
+            new_date = next_date + relativedelta(days=interval - 1)
+        elif contract.recurring_rule_type == 'weekly':
+            new_date = next_date + relativedelta(weeks=interval, days=-1)
+        else:
+            new_date = next_date + relativedelta(months=interval, days=-1)
```

Comment:
It can't be done because the relativedelta contains a variable interval depending on the contract, and I think that nest the dictionary inside the method is uglier.